src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,252 @@
+
+interface Column {
+    name: string;
+    alias?: string;
+    align?: TableCellProps["align"];
+    format?: (value: any, row: RowData) => any;
+    footerFormat?: (value: any, row: RowData) => any;
+    parse?: (value: any, row: RowData, column: Column) => any;
+    columnCalculation?: "sum" | "average" | ((values: any) => any);
+    editable?: boolean;
+    type?: "text" | "number" | string;
+}
+
+interface EditableTableElementProps
+    extends FormElementProps<RowData[]>,
+        SettableBoxProps,
+        SettableTableProps {
+    cols: Column[];
+    rows: RowData[];
+    footerLabel?: string;
+    onMouseEnter?: (row: RowData) => void;
+    onMouseLeave?: (row: RowData) => void;
+    rowCalculation?: (data: RowData[], row: RowData) => void;
+}
+
+//Mimic MUI header style for footer
+const TableCellStyleOverrides: TableCellProps["sx"] = {
+    fontFamily: "var(--defaultFont)",
+    backgroundColor: "var(--secondaryBackground)",
+    fontWeight: "bold",
+    color: "var(--secondaryForeground)",
+};
+
+const calculateFooterRow = (cols: Column[], rows: RowData[]): FooterRow => {
+    const totalCols = cols.filter((x) => x.columnCalculation);
+    const total: FooterRow = {};
+    for (const col of totalCols) {
+        if (col.columnCalculation) {
+            const values = rows.map((x) => x[col.name]);
+            if (col.columnCalculation === "sum") {
+                total[col.name] = calculateSum(values);
+            } else if (col.columnCalculation === "average") {
+                total[col.name] = calculateAverage(values);
+            } else {
+                total[col.name] = col.columnCalculation(values);
+            }
+        }
+    }
+    return total;
+};
+
+const calculateSum = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    return numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+};
+
+const calculateAverage = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    const sum = numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+    return sum / numericVals.length || 0;
+};
+/**
+ * A table form element.
+ * @displayName Editable Table
+ * @description A table form element that allows for inline editing and tabular calculation.
+ * @param props The props that will be provided by the Workflow runtime.
+ */
+function EditableTableElement(props: EditableTableElementProps): React.ReactElement | null {
+    const {
+        rows,
+        cols,
+        maxHeight,
+        maxWidth,
+        onMouseEnter,
+        onMouseLeave,
+        raiseEvent,
+        setValue,
+        size,
+        stickyHeader,
+        footerLabel,
+    } = props;
+    const [data, setData] = useState(rows);
+    setValue(data);
+    useEffect(() => {
+        setData(rows);
+    }, [rows]);
+
+    const illegalCols = cols.filter(
```

Comment:
This works, but given that this is a React component that is going to call this a lot, you are going to get huge numbers of console logs.

I would just add `const type = col.type === "number" ?  "number" : "text";` around line 190.